rustls/src/error.rs

```diff
@@ -1,11 +1,40 @@
-use crate::msgs::enums::{AlertDescription, ContentType, HandshakeType};
+use crate::msgs::enums::{AlertDescription, ContentType, HandshakeType, KeyUpdateRequest};
 use crate::rand;
 
 use std::error::Error as StdError;
 use std::fmt;
 use std::sync::Arc;
 use std::time::SystemTimeError;
 
+/// A corrupt TLS message payload that resulted in an error.
+#[non_exhaustive]
+#[derive(Debug, Clone)]
+pub enum InvalidMessage {
+    /// The peer sent us a syntactically incorrect TLS message.
```

Comment:
Just from a user's point of view, what would be the error message when sending a starttls message to a TLS server or vice-versa? Is there a way to make it painfully clear in the error message?

Reply:
If you send a message like `STARTTLS` to a TLS server, rustls on the server would yield an `InvaidMessage::InvalidContentType`, because it doesn't find a known content type in the range 0x14..=0x18 in the first byte. That seems pretty reasonable?

For "vice-versa", rustls wouldn't get invoked if you send TLS messages to a peer expecting a plaintext byte stream, but TLS messages are fairly recognizable from their 5 byte header, starting with the `ContentType` byte, then a two-byte version (rustls accepts 0x03XX at the message decoding level), then a two-byte length.

Reply:
I don't think we have a straightforward route to decorating these error variants with more detailed context.